src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -62,6 +62,28 @@ createFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf coup
   prepareAndTagClaims couponClaims "Fix rate payment"
 -- FIXED_RATE_BOND_COUPON_CLAIMS_END
 
+-- | Calculate a fix rate amount (if a credit event has not yet happened) for each payment date and create claims.
+createConditionalCreditFixRatePaymentClaims : Applicative f => [SchedulePeriod] -> PeriodicSchedule -> Bool -> Decimal -> Decimal -> DayCountConventionEnum -> Deliverable -> Observable -> f [TaggedClaim]
+createConditionalCreditFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf couponRate weightMultiplier dayCountConvention cashInstrumentCid defaultProbabilityReferenceId = do
+  let
+    creditEvent = Lte (Const 1.0, Observe defaultProbabilityReferenceId)
+    couponDatesAdjusted = map (.adjustedEndDate) schedule
+    couponAmounts = map (\p -> couponRate * (calcPeriodDcf dayCountConvention p useAdjustedDatesForDcf periodicSchedule.terminationDate periodicSchedule.frequency)) schedule
+    couponClaims = zipWith (\d a -> when (TimeGte $ d) $ scale (Const weightMultiplier * Const a) $ one cashInstrumentCid) couponDatesAdjusted couponAmounts
```

Comment:
The fixed coupon stream can probably be factored out into a helper function, and reused across bonds and swaps. But happy for this to go into a different PR.

Reply:
I agree, I already have a ticket for that: https://github.com/digital-asset/daml-finance/issues/149 (item 4).
This becomes even easier now that `until creditEvent` is separated to the top level.